### ISSUE TITLE
Decouple Logout Mutation from "Block Unauthorized Domains" Setting

### ISSUE THREAD

**Issue by alexookah** [AxeWP/wp-graphql-headless-login#158]
The current implementation of the logout mutation relies on the "Block Unauthorized Domains" setting, which introduces challenges for local development and environments where CORS is not applicable, such as CLI or server-to-server requests. While blocking unauthorized domains enhances security by requiring an origin header, this approach can inadvertently hinder valid use cases where the origin is not available or spoofable.

With the recent settings refactor, it is now feasible to decouple functionalities like logout mutation from CORS-based restrictions. For example, the logout mutation could depend on whether a Provider is configured to use an authentication cookie instead of strictly relying on origin validation.

Suggested Improvements
Update the logout mutation to function independently of the "Block Unauthorized Domains" setting.
Introduce a "Strict Mode" toggle for production environments to enforce stricter security measures while easing restrictions for local development.
Implement a mechanism to gate CORS-related behavior without impacting non-browser environments (e.g., CLI, GraphQL Codegen).
Context
This issue arises primarily due to the differences between browser and non-browser environments in handling CORS and the origin header. By addressing this, we can improve developer experience and better support multiple environment profiles without compromising production security.

Additional Notes
As part of future work on supporting multiple environment profiles, revisiting this coupling would be an ideal opportunity to enhance the plugin’s flexibility.

**Comment by justlevine** [comment]
In the interim, the `graphql_login_cookie_setting` filter can be used to programmatically enable/disable settings.

So you could force-enable the logout mutation with
```php
add_filter( `graphql_login_cookie_setting`,
  static function( $value, string $option_name ) {
    // Force-enable the logout mutation.
    if ( 'hasLogoutMutation' === $option_name ) {
      return true;
    }

   return $value; // all other options
},
10, 2 );
```

Now mind, there isn't a session or a cookie to clear without `Access-Control-Allow-Credentials`, and that only works with `Access-Control-Allow-Orign`, but it will keep the schema stable in environments where an `Origin` header can't even be set manually.

**Comment by justlevine** [comment]
## 🤖 Automated Triage

**Classification:** type: enhancement ⚡ · priority: low · status: needs discussion 💬
**Agent-feasible:** Partially — the core decoupling (gating logout on cookie-auth vs. origin check) is well-scoped and implementable, but the "Strict Mode" toggle design needs a product decision first.
**Estimated effort:** Agent medium · Human medium

**Summary:** The logout mutation currently depends on the "Block Unauthorized Domains" CORS check, which breaks CLI/server-to-server and local dev flows. With the settings refactor landed, logout can be gated on whether the Provider uses an auth cookie instead of origin validation. The "Strict Mode" toggle concept needs design alignment before implementation.

**Next steps:** Needs human decision on whether to implement a "Strict Mode" toggle or simply decouple logout from CORS checks entirely. Once the approach is settled, this is ready for the planning loop.
